### PR TITLE
Organize logging levels

### LINKED ISSUE #3856: Reduce logging to Sentry
We're error logging a number of things that shouldn't be error logs. This pollutes Sentry with a number of issues that aren't really our issue (build failures on pip/etc). We should clean up our logging here to reduce noise and get back to a reasonable number of logging events.

### PR DESCRIPTION
**Pull request by humitos**
Log only what we need as ERROR when it's something that we need/want to take a look that something could be a bug/issue.

There are some ERROR that were replaced by WARNING since they are some failures but that we don't need to take a look and the message shown to the user should be enough for them.

Check #3898 for style changes

### CODE REVIEW

**Comment by davidfischer** [comment]
I'll take this one on however the CI issues look real.

**Comment by humitos** [comment]
@davidfischer I didn't understand your comment and why you assigned this PR to you :P , but I just fixed the linting issues reported by travis.

File: readthedocs/builds/syncers.py [hunk 1]
```diff
@@ -58,8 +58,7 @@ def copy(cls, path, target, is_file=False, **__):
                 mkdir_cmd = ("ssh %s@%s mkdir -p %s" % (sync_user, server, target))
                 ret = os.system(mkdir_cmd)
                 if ret != 0:
-                    log.info("COPY ERROR to app servers:")
-                    log.info(mkdir_cmd)
+                    log.error("Copy error to app servers: cmd=%s", mkdir_cmd)
```
Comment (stsewd) [inline readthedocs/builds/syncers.py]: Can we start using single quotes `'` around all the strings? :pray:

File: readthedocs/projects/tasks.py [hunk 2]
```diff
@@ -183,12 +183,15 @@ def run(self, version_pk):  # pylint: disable=arguments-differ
             self.project = self.version.project
             self.sync_repo()
             return True
-        # Catch unhandled errors when syncing
+        except RepositoryError as e:
+            # Do not log as ERROR handled exceptions
+            log.warning('There was an error with the repository: msg=%s', e.msg)
```
Comment (davidfischer) [inline readthedocs/projects/tasks.py]: We could pass `exc_info=True` to `log.warning` if you want the full exception.

Comment (humitos) [reply readthedocs/projects/tasks.py]: You are right. Changing.

**Pull request merged**